aderyn_core/src/detect/high/tautology_or_contradiction.rs

```diff
@@ -0,0 +1,608 @@
+
+        None
+    }
+
+    /// Does this make sense? (boolean answer)
+    /// ```no_code
+    ///     if(value, operator, uint8?)
+    /// ```
+    ///
+    /// Take advantage of the above method by reusing code
+    ///
+    /// Example (ways to call this function)
+    ///
+    /// Say x is uint8:
+    /// Then, when we come across a binary operation like follows:
+    ///     300 >= x
+    /// we can determine if it makes sense by calling this function
+    ///     does_operation_make_sense_with_lhs_value("300", ">=", "uint8")
+    ///
+    /// Notice, here the value 300 is on the left hand side.
+    ///
+    pub fn does_operation_make_sense_with_lhs_value(
+        value: &str,
+        operator: &str,
+        type_string: &str,
+    ) -> Option<bool> {
+        let inverse_operator = {
+            match operator {
+                ">=" => Some("<="),
+                "<=" => Some(">="),
+                ">" => Some("<"),
+                "<" => Some(">"),
+                _ => None,
+            }
+        }?;
+        does_operation_make_sense_with_rhs_value(type_string, inverse_operator, value)
+    }
+
+    /// Accept the type string to calculate the range.
+    pub fn get_range_for_type_string(
+        type_string: &str,
+    ) -> Result<SolidityNumberRange, Box<dyn Error>> {
+        if type_string.starts_with("uint") {
+            if let Some((_, num_of_bits)) = &type_string.split_once("uint") {
+                let num_of_bits = num_of_bits.parse::<u32>()?;
+                return Ok(SolidityNumberRange {
+                    min_val: find_uint_min(num_of_bits),
+                    max_val: find_uint_max(num_of_bits),
+                });
+            }
+        } else if type_string.starts_with("int") {
+            if let Some((_, num_of_bits)) = &type_string.split_once("int") {
+                let num_of_bits = num_of_bits.parse::<u32>()?;
+                return Ok(SolidityNumberRange {
+                    min_val: find_int_min(num_of_bits),
+                    max_val: find_int_max(num_of_bits),
+                });
+            }
+        }
+        Err("Invalid type string provided!".into())
+    }
+
+    // Helpers to calculate min and max for uint types like uint8, uint16, uint24, and so on . . .
+
+    fn find_uint_max(num_of_bits: u32) -> BigInt {
+        BigInt::parse_bytes(b"2", 10).unwrap().pow(num_of_bits) - BigInt::one()
+    }
+
+    fn find_uint_min(_: u32) -> BigInt {
+        BigInt::ZERO
+    }
+
+    // Helpers to calculate min and max for int types like int8, int16, int24, and so on . . .
+
+    fn find_int_max(num_of_bits: u32) -> BigInt {
+        BigInt::parse_bytes(b"2", 10).unwrap().pow(num_of_bits - 1) - BigInt::one()
+    }
+
+    fn find_int_min(num_of_bits: u32) -> BigInt {
+        BigInt::parse_bytes(b"2", 10)
+            .unwrap()
+            .pow(num_of_bits - 1)
+            .neg()
+    }
+
+    #[cfg(test)]
+    mod test_num_bigint_primitives {
+
+        use std::ops::Neg;
+
```

Comment:
@alexroan I was wondering if you had a chance to look at these tests? I'm happy to write more in the Solidity file but .. I just want to make sure you saw these ones